whois-nrtm4/src/main/java/net/ripe/db/nrtm4/DeltaChange.java

```diff
@@ -0,0 +1,57 @@
+package net.ripe.db.nrtm4;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+
+
+public class DeltaChange {
+
+    enum Action {
+        DELETE,
+        ADD_MODIFY;
+
+        @Override
+        @JsonValue
+        public String toString() {
+            return name().toLowerCase();
+        }
+    }
+
+    private final Action action;
+    @JsonProperty("object_class")
+    private final ObjectType objectClass;
+    @JsonProperty("primary_key")
+    private final String primaryKey;
+    private final RpslObject object;
+
```

Comment:
You do not need to add a custom serializer in payloadProcessor, we can get rid of that class all together ? You can directly put annotation  here and write a converter here  @JsonSerialize(converter = RplsObjectSerializer.class).  You need to extend StdConverter.

Reply:
i tried that approach earlier but was getting NPE from Jackson on the objectId field. i'm not sure how to get it working coz the config looks ok to me. can we come back to this when we have a solution? it has a very low impact on the code.